src/components/ActorList.jsx

```diff
@@ -3,39 +3,37 @@ import React, {useState} from 'react';
 const ActorList = ({actors, setActors}) => {
   const [deleteError, setDeleteError] = useState('');
 
-
-  const deleteActorHandler = (event) => {
+  const deleteActorHandler = async (event) => {
     const id = event.target.value
-    fetch('/actors/' + id, {
-      method: 'DELETE',
-    })
-      .then((response) => {
-        if (response.ok) {
-          // upon successful deletion, remove that actor's name from the actor list
-          setDeleteError('');
-          setActors(actors.filter((actor) => actor.id.toString() !== id));
-        } else {
-          throw new Error('Error when deleting the actor.');
-        }
-      })
-      .catch((error) => {
-        setDeleteError(error.message);
-      });
+
+    try {
+      const response = await fetch('/actors/' + id, { method: 'DELETE' });
+
+      if (response.ok) {
+        // upon successful deletion, remove that actor's name from the actor list
+        setDeleteError('');
+        setActors(actors.filter((actor) => actor.id.toString() !== id));
+      } else {
+        setDeleteError('Error when deleting the actor.');
+      }
+    } catch (error) {
+      setDeleteError(error.message);
+    }
```

Comment:
If the error that will be caught with the catch is a network error, would we rather console.error and give a more ambiguous error to the user? I know we haven't set a standard here for that but since the current structure can more easily identify the two error types, maybe we should discuss what the standard approach should be.

Reply:
Agreed, but this is probably something to look into later